src/js/tech/flash.js

```diff
@@ -21,21 +21,13 @@ let navigator = window.navigator;
  */
 class Flash extends Tech {
 
-  constructor(player, options, ready){
-    super(player, options, ready);
+  constructor(options, ready){
+    super(options, ready);
```

Comment:
`super` is Component and would still require player as the first arg. In order to make this happen we would probably need to update Component to be a generic UI component class that doesn't require a player, and then create a PlayerComponent subclass that does. That will take some work, so we might want to save that for the end, or even a secondary PR after we finish everything else here.